### PR TITLE
Carry #18912 : Docker Remote API documentation update

### PR DESCRIPTION
**Pull request by vdemeester**
Carry #18912. Taken into account @jamtur01 and @moxiegirl comments.

🐸

/cc @jamtur01 @thaJeztah @moxiegirl @MHBauer 

---

I was confused for the longest time on how to actually use and make requests against the remote API, so I think that it might help for those getting started with it to know how to actually test it out via curl. I added in parts on how to access the remote API via curl against the default unix socket, and also on how to configure the docker daemon to expose the API on a TCP port as well if desired.

Signed-off-by: Michelle Liu michelle@michelleliu.io

### CODE REVIEW

**Comment by MHBauer** [comment]
Something weird happened I think. It looks like the first paragraph is repeated.

**Comment by vdemeester** [comment]
@MHBauer oh you're right, my bad :angel:

File: docs/reference/api/docker_remote_api.md [hunk 1]
```diff
@@ -24,6 +24,20 @@ client must have `root` access to interact with the daemon. If a group named
...
 
+To interact with the remote API using [cURL](http://curl.haxx.se/download.html).
+You run `curl` against the socket where the Docker daemon is listening. For this
+reason, must use cURL 7.40 (or greater) as these versions have the
+`--unix-socket` flag available. To run `curl` against the deamon on
+the default socket, use the following:
+
+    curl --unix-socket /var/run/docker.sock http://containers/json
+
+If you bound the Docker daemon to a different, you would reference that
```
Comment (MHBauer) [inline docs/reference/api/docker_remote_api.md]: bound to a different _what_? Port doesn't really make sense here. I don't think unix sockets have ports.

File: docs/reference/api/docker_remote_api.md [hunk 2]
```diff
@@ -24,6 +24,20 @@ client must have `root` access to interact with the daemon. If a group named
...
+alternative port in your cURL rather than the default.
+
+Please note that exposing the Remote API can be a security issue and
+you should consider this before binding the API.
```
Comment (MHBauer) [inline docs/reference/api/docker_remote_api.md]: Seems random to have this warning here. I agree a warning is needed somewhere. I'm not sure where.

Comment (vdemeester) [inline docs/reference/api/docker_remote_api.md]: You're right. Pretty sure it is somewhere else already even.

**Comment by vdemeester** [comment]
@MHBauer updated.

**Comment by thaJeztah** [comment]
LGTM

**Comment by icecrime** [comment]
Ping @moxiegirl @MHBauer!

File: docs/reference/api/docker_remote_api.md [hunk 3]
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
 `docker` exists on your system, `docker` applies ownership of the socket to the
 group.
 
+To interact with the remote API using [cURL](http://curl.haxx.se/download.html).
```
Comment (tiborvass) [inline docs/reference/api/docker_remote_api.md]: `https`

Comment (vdemeester) [inline docs/reference/api/docker_remote_api.md]: oh nooeess :scream:

**Comment by vdemeester** [comment]
@tiborvass updated :wink:

**Comment by tiborvass** [comment]
LGTM

File: docs/reference/api/docker_remote_api.md [hunk 4]
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
...
 
+To interact with the remote API using [cURL](https://curl.haxx.se/download.html).
+You run `curl` against the socket where the Docker daemon is listening. For this
+reason, must use cURL 7.40 (or greater) as these versions have the
+`--unix-socket` flag available. To run `curl` against the deamon on
+the default socket, use the following:
+
+    curl --unix-socket /var/run/docker.sock http://containers/json
+
+If you bound the Docker daemon to a different socket path or TCP port,
```
Comment (tiborvass) [inline docs/reference/api/docker_remote_api.md]: shouldnt this be `bind`?

**Comment by tiborvass** [comment]
Need a native-speaker LGTM @MHBauer @jamtur01

File: docs/reference/api/docker_remote_api.md [hunk 5]
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
 `docker` exists on your system, `docker` applies ownership of the socket to the
 group.
 
+To interact with the remote API using [cURL](https://curl.haxx.se/download.html).
```
Comment (jamtur01) [inline docs/reference/api/docker_remote_api.md]: This is either a sentence fragment or a header? Unclear.

File: docs/reference/api/docker_remote_api.md [hunk 6]
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
...
+You run `curl` against the socket where the Docker daemon is listening. For this
+reason, must use cURL 7.40 (or greater) as these versions have the
```
Comment (jamtur01) [inline docs/reference/api/docker_remote_api.md]: To connect to the Docker daemon with `cURL` you need to use version 7.40 or later, as these versions provide the `--unix-socket` flag.

File: docs/reference/api/docker_remote_api.md (hunk 4 above)
Comment (jamtur01) [inline docs/reference/api/docker_remote_api.md]: If you have bound the ...

File: docs/reference/api/docker_remote_api.md [hunk 7]
```diff
@@ -24,6 +24,17 @@ client must have `root` access to interact with the daemon. If a group named
...
+`--unix-socket` flag available. To run `curl` against the deamon on
+the default socket, use the following:
+
+    curl --unix-socket /var/run/docker.sock http://containers/json
+
+If you bound the Docker daemon to a different socket path or TCP port,
+you would reference that later in your cURL rather than the default.
```
Comment (jamtur01) [inline docs/reference/api/docker_remote_api.md]: -later

**Comment by jamtur01** [comment]
Needs some rewrites I think.

**Comment by vdemeester** [comment]
updated ^_^

**Comment by thaJeztah** [comment]
thanks, still LGTM

**Pull request merged**